### PR TITLE
[JAVA] Add micronaut framework

### PR DESCRIPTION
**Pull request by gnomeria**
Before submitting your PR, please review the following checklist :

## If you are adding a framework

+ [x] A `Dockerfile` exists, or can be built ?
+ [x] I've updated `CI` config file (and **commited** `.travis.yml`)
~~~sh
bin/make ci_config
~~~
+ [x] I've updated `.dependabot` config file  (and **commited** `.dependabot/config.yml`)
~~~sh
bin/make deps_config
~~~
+ [x] All tests passes ?
~~~
export FRAMEWORK=micronaut
shards install
shards build --release
bin/make config
bin/neph ${FRAMEWORK}
crystal spec
~~~

### CODE REVIEW

**Comment by waghanza** [comment]
why closed ? any contribution is valuable

**Comment by gnomeria** [comment]
I still need to update the ci build etc I guess. I'll reopen it later when I'm done @waghanza.

**Comment by waghanza** [comment]
@gnomeria you could re-open, i'll help (make a `PR` to merge in this) if needed :heart:

**Comment by gnomeria** [comment]
Also my project was currently in gradle, does the dependabot support it?

**Comment by waghanza** [comment]
@gnomeria I am surely not able to determine which is better for your needs, `maven` or `gradle`
what I can say is that `here` we use `gradle` for `kotlin` and `maven` for `java`, since it is the most common use case

however, if you prefer to use `gradle` in can turn this project using `bot`. I just need a functional implementation :stuck_out_tongue:

File: java/micronaut/config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+framework:
+  website: micronaut.io
+  version: 1.2.5
+
+image: gradle
```
Comment (waghanza) [inline java/micronaut/config.yaml]: should be 
```yaml
deps:
  - gradle
```

I have made a slight modification to code base so as to run `java` frameworks on `alpine`

File: java/micronaut/config.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+framework:
+  website: micronaut.io
+  version: 1.2.5
```
Comment (waghanza) [inline java/micronaut/config.yaml]: it should be 1.2, using X.Y.Z is too heavy, in term of maintenance

File: java/micronaut/gradle.properties [hunk 3]
```diff
@@ -0,0 +1 @@
+micronautVersion=1.2.5
```
Comment (waghanza) [inline java/micronaut/gradle.properties]: could it be a range ?
using `1.2.5` is too strict, I do not know `gradle`, but I think using
```ini
micronautVersion=1.2.+
```
should work as expected

Comment (gnomeria) [reply java/micronaut/gradle.properties]: sure i'll update it

File: java/micronaut/config.yaml (hunk 1 above)
Comment (gnomeria) [reply java/micronaut/config.yaml]: so i just need to put as :
```
image: gradle
deps:
  - gradle
```

**Comment by waghanza** [comment]
@gnomeria could you run
~~~sh
shards install
shards build
bin/make ci_config
bin_make deps_config
~~~
so as to resolve conflicts (`.travis.yml` and `.dependabot/config.yml` are generated) ?

**Comment by waghanza** [comment]
@gnomeria Still failing, since `gradle` version is **4.10.3** on `alpine`

I'll fix that tomorrow

**Comment by gnomeria** [comment]
> @gnomeria Still failing, since `gradle` version is **4.10.3** on `alpine`
> 
> I'll fix that tomorrow

I forgot about the gradle version. Previously I thought I could just define my own `Dockerfile`, so I just install the gradle version on the alpine by:
```bash
ENV GRADLE_VERSION 5.5
# get gradle and supporting libs
RUN apk -U add --no-cache curl; \
  curl https://downloads.gradle.org/distributions/gradle-${GRADLE_VERSION}-bin.zip > gradle.zip; \
  unzip gradle.zip; \
  rm gradle.zip; \
  apk del curl; \
  apk update && apk add --no-cache libstdc++ && rm -rf /var/cache/apk/*
ENV PATH=${PATH}:/gradle-${GRADLE_VERSION}/bin
```

**Comment by waghanza** [comment]
> Previously I thought I could just define my own Dockerfile

Could be that, but I try to have only `Dockerfile` per language, in order to keep run _fair_ (I mean running on the same version of **language** ...)

**Pull request merged**